test/usecase/get_surveys_use_case_test.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/api/response/base/base_http_response.dart';
+import 'package:flutter_survey/api/response/survey_response.dart';
+import 'package:flutter_survey/models/survey.dart';
+import 'package:flutter_survey/usecase/base/base_use_case.dart';
+import 'package:flutter_survey/usecase/get_surveys_use_case.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../mock/mock_data.mocks.dart';
+import '../utils/file_util.dart';
+
+void main() {
+  group('GetSurveysUseCaseTest', () {
+    late MockSurveyRepository mockRepository;
+    late GetSurveysUseCase getSurveysUseCase;
+    late Map<String, dynamic> surveysJson;
+
+    setUp(() async {
+      mockRepository = MockSurveyRepository();
+      getSurveysUseCase = GetSurveysUseCase(mockRepository);
+      surveysJson = await FileUtil.loadFile('test_resources/surveys.json');
+    });
+
+    test('When calling API with valid data, it returns Success result',
+        () async {
+      final surveyResponses =
+          BaseHttpResponseList<SurveyResponse>.fromJson(surveysJson);
+      final surveys = surveyResponses.data
+          .map((response) => response.attributes)
+          .toList()
+          .map((response) => response.toSurvey())
+          .toList();
+
+      when(mockRepository.getSurveys(any, any))
+          .thenAnswer((_) async => surveys);
```

Comment:
at this stage, I would rather create another surveys.json file that has this direct data node instead of reuse and reparse the same old data in this test file; later in a real project, it will just get to be too complex to understand your test I don't think it's a good test then. Test should be easy to write and understand, if you exercise on a mock data set, better provide it straight away rather than performing at the same effort as you are writing production code to reproduce a set of mock data 👌  just a suggestion to move forward, better save your time for something else and make it easy for later debug lol

Reply:
I think an empty Survey list is fine in this case to verify Success case 🤔

Reply:
that's true, if you had tested the parser from SurveyResponse -> Survey somewhere else then in here you don't really need to care about the details